File: backend/internal/types/status.go
package types

import "errors"

type Status string

const (
	Unstarted Status = "unstarted"
	Started   Status = "started"
	Finished  Status = "finished"
)

func ValidateStatus(status Status) error {
	switch status {
	case Unstarted, Started, Finished:
		return nil
	default:
		return errors.New("status invalido")
	}
}

File: frontend/src/utils/functions/status.ts
import type { Status } from "../../types";

export function getPreviousStatus(status: Status): Status | null {
  if (status === "finished") return "started";
  if (status === "started") return "unstarted";
  return null;
}

export function getNextStatus(status: Status): Status | null {
  if (status === "unstarted") return "started";
  if (status === "started") return "finished";
  return null;
}
